bmaptools/BmapCopy.py

```diff
@@ -211,6 +212,28 @@ def __init__(self, image, dest, bmap=None, image_size=None):
 
         self._batch_blocks = self._batch_bytes // self.block_size
 
+    def set_psplash_pipe(self, pipe):
+        """
+        Set the psplash named pipe file path to be used when updating the
+        progress - best effort.
+
+        The 'pipe' argument is the named pipe used by the psplash process to get
+        progress bar commands. When the pipe argument doesn't exist or is not a
+        pipe, the function will ignore with a warning. This behavior is
+        considered as such because the progress is considered a decoration
+        functionality which might or might not be available even if requested.
+        When used as a boot service, the unavailability of the psplash service
+        (due to various reasons: no screen, racing issues etc.) should not
+        break the writting process. This is why this implementation is done as
+        a best effort.
+        """
+
+        if os.path.exists(pipe) and stat.S_ISFIFO(os.stat(pipe).st_mode):
+            self._psplash_pipe = pipe
+        else:
+            _log.warning("%s is not a pipe so psplash progress will not be "
```

Comment:
Just to be consistent with the rest of the code, please, use single quotes for file names ('%s' instead of %s).